pkg/etcdcli/etcdcli.go

```diff
@@ -411,16 +404,25 @@ func (g *etcdClientGetter) MemberStatus(ctx context.Context, member *etcdserverp
 	return EtcdMemberStatusAvailable
 }
 
+// Defragment creates a new uncached clientv3 to the given member url and calls clientv3.Client.Defragment.
 func (g *etcdClientGetter) Defragment(ctx context.Context, member *etcdserverpb.Member) (*clientv3.DefragmentResponse, error) {
-	cli, err := g.getEtcdClientWithClientOpts([]string{member.ClientURLs[0]}, WithDialTimeout(DefragDialTimeout))
+	// no g.clientLock necessary, this always returns a new fresh client
+	cli, err := newEtcdClientWithClientOpts([]string{member.ClientURLs[0]}, WithDialTimeout(DefragDialTimeout))
 	if err != nil {
 		return nil, fmt.Errorf("failed to get etcd client for defragment: %w", err)
 	}
-	defer cli.Close()
+	defer func() {
+		if cli == nil {
+			return
+		}
+		if err := cli.Close(); err != nil {
```

Comment:
do we call close or Return() to give it back to the sync.Pool ?

Reply:
as the comment on the func says: 
>// Defragment creates a new uncached clientv3 to the given member url and calls clientv3.Client.Defragment.

I don't know why it did not use the cache client before, I assume due to single endpoint target nature of that method. But Close() is correct here.

Reply:
sounds good 👍🏽